python/ray/air/config.py

```diff
@@ -146,34 +145,44 @@ class DatasetConfig:
     split: Optional[bool] = None
 
     # Whether to raise an error if the Dataset isn't provided by the user.
-    # True by default for the "train" dataset only.
+    # False by default.
     required: Optional[bool] = None
 
-    # Whether the dataset can be streamed into memory using pipelined reads.
-    # When enabled, get_dataset_shard() returns DatasetPipeline instead of Dataset.
-    # Note that streaming isn't enabled unless you set stream_window_size too.
-    # False by default for all datasets.
-    streamable: Optional[bool] = None
-
     # Whether to transform the dataset with the fitted preprocessor. This must be
     # enabled at least for the dataset that is fit.
-    # True by default for all datasets.
+    # True by default.
     transform: Optional[bool] = None
 
-    # List of fields that the user cannot override in ``dataset_config``.
-    _noncustomizable_fields: Tuple[str] = ()
```

Comment:
why remove `_noncustomizable_fields`?

Reply:
It's handled via trainer validate_attributes() now.